### ISSUE TITLE
Undefined behavior found when serializing zero-sized "item"

### ISSUE THREAD

**Issue by bjadamson** [eyalz800/zpp_bits#112]
Hey, first of all thank you so much for this library! I really enjoyed your talk on youtube about it as well. :)

Today I decided to start using undefined behavior sanitizer for my project and found one hit from zpp_bits, here:
https://github.com/eyalz800/zpp_bits/blob/main/zpp_bits.h#L2621-L2623

It turns out passing nullptr to mempcy is undefined behavior, in my distro's string.h, memcpy is annotated with non_null:

```C
/* Copy N bytes of SRC to DEST.  */
extern void *memcpy (void *__restrict __dest, const void *__restrict __src,

		     size_t __n) __THROW __nonnull ((1, 2));
```

For completeness sake here's a screenshot from me debugging the issue locally:
![ub zpp_bits](https://user-images.githubusercontent.com/1087837/236262467-1e697a8f-f7e0-43ff-b05f-bb52b9a138b8.png)

I'm guessing the solution is to just not call memcpy if the `item_size_in_bytes == 0`, but I'm not sure if that means you can just do nothing in that case or what.

**Comment by bjadamson** [comment]
Some more discussion I found on whether NULL is considered an invalid value when calling memcpy:
https://stackoverflow.com/a/5243068

**Comment by eyalz800** [comment]
Thanks for reporting this - https://github.com/eyalz800/zpp_bits/issues/112

**Comment by eyalz800** [comment]
Fixed v4.4.15.
